### PR TITLE
[Practice] Surgery

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request
Creates a button that can be used to spawn humans with 3 different conditions to practice surgery.
Adds 6 small surgery rooms to the admin level that have the button.

## Why It's Good For The Game
having the ability to have a button to practice surgery will help with those who need practice and training.

## Changelog
:cl:
add: Added a practice surgery button
add: Added 4 practice surgery rooms to MarineHQ
/:cl:

### CODE REVIEW

File: code/game/area/centcom.dm [hunk 1]
```diff
@@ -59,3 +59,10 @@
 
 /area/centcom/aitest //Part of the centcom zlevel, basically an arena for AI that ultilizes nodes
 	name = "Artificial Intelligence Test Zone"
+
+// Practice Areas
+/area/centcom/practice
+	name = "\improper Practice Center"
+
+/area/centcom/practice/medical
+	name = "\improper Practice Medical Center"
```
Comment (psykzz) [inline code/game/area/centcom.dm:68]:
```suggestion
	name = "\improper Medical Practice Center"
```

File: code/game/objects/machinery/practice/medical.dm [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	resistance_flags = RESIST_ALL
+	var/mob/living/carbon/human/humanspawned = null
+
+/obj/machinery/practice/medical/surgery/attack_hand(mob/living/user)
+	if(user.a_intent == INTENT_HARM)
+		to_chat(user, "<span class='warning'>You are unable to damage the button.</span>")
+		return
+	if(humanspawned)
+		qdel(humanspawned)
+		humanspawned = null
```
Comment (psykzz) [inline code/game/objects/machinery/practice/medical.dm:15]:
```suggestion
		QDEL_NULL(humanspawned)
```

**Comment by psykzz** [comment]
👍 good start
Though the idea was to have these on the MarineHQ map, before you get too far you may want to find a spot and do the mapping on there instead.

**Comment by jjpark-kb** [comment]
> 👍 good start
> Though the idea was to have these on the MarineHQ map, before you get too far you may want to find a spot and do the mapping on there instead.

Ah, okay, Ill do that!

File: _maps/map_files/MarineHQ/MarineHQ.dmm [hunk 3]
```diff
@@ -4709,6 +4758,15 @@
 /obj/structure/cable,
 /turf/open/floor/wood,
 /area/mainship/command/cic)
+"vz" = (
+/obj/machinery/shower{
+	dir = 4;
+	pixel_y = -3
```
Comment (psykzz) [inline _maps/map_files/MarineHQ/MarineHQ.dmm:4764]:
```suggestion
	dir = 4
```

File: _maps/map_files/MarineHQ/MarineHQ.dmm [hunk 4]
```diff
@@ -4728,6 +4791,31 @@
 	dir = 2
 	},
 /area/mainship/shipboard/firing_range)
+"vO" = (
+/obj/machinery/reagentgrinder,
+/obj/item/stack/sheet/mineral/phoron{
+	amount = 25;
+	pixel_x = 3;
+	pixel_y = 3
+	},
```
Comment (psykzz) [inline _maps/map_files/MarineHQ/MarineHQ.dmm:4800]:
```suggestion
/obj/item/stack/sheet/mineral/phoron,
```

Comment (psykzz) [reply _maps/map_files/MarineHQ/MarineHQ.dmm:4800]: don't do this

File: _maps/map_files/MarineHQ/MarineHQ.dmm [hunk 5]
```diff
@@ -8213,6 +8444,29 @@
...
+/obj/item/stack/sheet/mineral/phoron{
+	amount = 25;
+	pixel_x = 3;
+	pixel_y = 3
+	},
+/turf/open/floor/mainship/sterile/corner,
+/area/centcom/practice/medical)
```
Comment (psykzz) [inline _maps/map_files/MarineHQ/MarineHQ.dmm:8469]:
```suggestion
/obj/structure/table/reinforced,
/obj/item/stack/sheet/mineral/phoron,
/obj/item/stack/sheet/mineral/phoron,
/turf/open/floor/mainship/sterile/corner,
/area/centcom/practice/medical)
```

File: _maps/map_files/MarineHQ/MarineHQ.dmm [hunk 6]
```diff
@@ -8927,6 +9348,18 @@
...
 /area/mainship/living/captain_mess)
+"Ze" = (
+/obj/machinery/optable,
+/obj/item/tank/anesthetic,
+/obj/item/clothing/mask/breath/medical,
+/obj/machinery/practice/medical/surgery{
+	dir = 8;
+	pixel_x = -22
+	},
+/turf/open/floor/mainship/sterile/side{
```
Comment (psykzz) [inline _maps/map_files/MarineHQ/MarineHQ.dmm:9359]:
```suggestion
/obj/machinery/practice/medical/surgery{
	dir = 8
	},
/turf/open/floor/mainship/sterile/side{
```

**Comment by psykzz** [comment]
I did these changes here so you understand what i meant before.
You'd need to remove the map edits from lavaland too.

**Pull request merged**